app/src/main/kotlin/com/mapbox/vision/teaser/SafetyFragment.kt

```diff
@@ -0,0 +1,238 @@
+package com.mapbox.vision.teaser
+
+import android.animation.Animator
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.view.animation.AccelerateDecelerateInterpolator
+import androidx.fragment.app.Fragment
+import com.mapbox.services.android.navigation.v5.navigation.NavigationConstants
+import com.mapbox.services.android.navigation.v5.utils.DistanceFormatter
+import com.mapbox.services.android.navigation.v5.utils.LocaleUtils
+import com.mapbox.vision.mobile.core.models.Country
+import com.mapbox.vision.mobile.core.models.detection.DetectionClass
+import com.mapbox.vision.safety.VisionSafetyManager
+import com.mapbox.vision.safety.core.VisionSafetyListener
+import com.mapbox.vision.safety.core.models.CollisionDangerLevel
+import com.mapbox.vision.safety.core.models.CollisionObject
+import com.mapbox.vision.safety.core.models.RoadRestrictions
+import com.mapbox.vision.teaser.models.UiSign
+import com.mapbox.vision.teaser.utils.SoundsPlayer
+import com.mapbox.vision.teaser.utils.classification.SignResources
+import com.mapbox.vision.teaser.utils.requireMainActivity
+import com.mapbox.vision.teaser.utils.requireVisionManager
+import com.mapbox.vision.teaser.utils.runOnUiThread
+import com.mapbox.vision.teaser.view.hide
+import com.mapbox.vision.teaser.view.show
+import kotlinx.android.synthetic.main.fragment_safety.*
+
+class SafetyFragment : Fragment() {
+
+    companion object {
+        val TAG: String = SafetyFragment::class.java.simpleName
+        private const val CALIBRATION_READY_VALUE = 1f
+        fun newInstance() = SafetyFragment()
+    }
+
+    private lateinit var soundsPlayer: SoundsPlayer
+    var calibrationProgress = 0F
+    var lastSpeed = 0f
+    var country = Country.Unknown
+    lateinit var signResources: SignResources
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
+        soundsPlayer = SoundsPlayer(requireContext())
+        return inflater.inflate(R.layout.fragment_safety, container, false)
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        val resources = requireMainActivity()?.signResources
+        if (resources == null) {
+            requireActivity().onBackPressed()
+            return
+        }
+        signResources = resources
+        calibration_progress.text = getString(R.string.calibration_progress, 0)
```

Comment:
nit: `setProgress(0)`
```
private fun setProgress(value: Int) {
   calibration_progress.text = getString(R.string.calibration_progress, value)
}
```

and use in `calibrationInProgress`

Reply:
fixed